### PR TITLE
Implemented unit tests 

### LINKED ISSUE #7: Develop automated tests
As discussed in #29, a http mocking library will be used to verify the requests are received and have the expected format (ie. the right piwik parameters).

For a more detailed explanation of the approach and its goals see https://github.com/piwik/piwik-dotnet-tracker/pull/46#issuecomment-287613257

### PR DESCRIPTION
**Pull request by ptr1120**
In order to fix #7:

* made some `protected` methods from PiwikTracker to `protected internal ` in order to access them from test assembly
* added test assembly as friend of piwik tracker assembly (in order to access `internal` methods)
* added nuget FakeHttpContext (https://github.com/vadimzozulya/FakeHttpContext) for mocking cookie / `HttpContext`
* added nuget SimpleHttpMock (https://github.com/xiaoyvr/SimpleHttpMock) for mocking piwik server
* implemented several tests

### CODE REVIEW

File: Piwik.Tracker/PiwikTracker.cs [hunk 1]
```diff
@@ -1516,7 +1524,7 @@ private string GetRequest(int idSite)
                     "&r=" + new Random().Next(0, 1000000).ToString("000000") +
 
                     // Only allowed for Super User, token_auth required,
-                    (!string.IsNullOrEmpty(_ip) ? "&cip=" + _ip : "") +
+                    (!string.IsNullOrEmpty(_ip) ? "&cip=" + UrlEncode(_ip) : "") +
```
Comment (julienmoumne) [inline Piwik.Tracker/PiwikTracker.cs]: Could you explain which problem did this solve?

Comment (ptr1120) [inline Piwik.Tracker/PiwikTracker.cs]: sorry, this I did not want to commit

**Comment by ptr1120** [comment]
Did you have a look at PiwikTrackerWithMockedServerTests.cs? There I did the discussed approach. Unit tests in PiwikTrackerTests do not need a mocked server.

**Comment by julienmoumne** [comment]
I find that comparing access logs would simplify a great deal the code.

We would have an expected access log such as
`127.0.0.1 - frank [10/Oct/2000:13:55:36 -0700] "GET /piwik.php?url=URL HTTP/1.0" 200 2326`

Which would be compared with the one produced by the mock http server.

In the mean time, let's merge the PR as it is as soon as it passes on travis.

You will need to submit a new PR as I unintentionally merged/reverted that one.

I suggest you create a branch on your repository and that you enable travis on it to verify tests are passing.

Thanks!

**Pull request merged**